profiling/src/profile/api.rs

```diff
@@ -120,3 +127,163 @@ pub struct Sample<'a> {
     /// things like a thread id, allocation size, etc
     pub labels: Vec<Label<'a>>,
 }
+
+pub struct Profile<'a> {
+    pub duration: Duration,
+    pub period: Option<(i64, ValueType<'a>)>,
+    pub sample_types: Vec<ValueType<'a>>,
+    pub samples: Vec<Sample<'a>>,
+    pub start_time: SystemTime,
+}
```

Comment:
Given the assumption

> There are no changes to shipped code, so there shouldn't be anything to test.

Should this be inside the `mod test`? And perhaps also document the assumption in the code, so that other/newer team members know that this isn't expected to be used outside of libdatadog testing.

Reply:
Maybe. I need to see what configuration the `bench` configuration uses -- I know it uses public APIs only, but I'm not sure if it builds with test or not, so that's the only reason I didn't immediately slap it into `#[cfg(test)]`.